app/workers/delete_object_hierarchy_worker.rb

```diff
@@ -56,14 +56,6 @@ def hierarchy_entries_for(ar_object)
         raise DoNotRetryError, "Background deleting #{ar_object.class}:#{ar_object.id} which is not destroyable."
       end
 
-      if ar_object.is_a?(FeaturesPlan)
-        # This is an ugly hack to handle lack of `#id` but we have only FeaturesPlans with a composite primary key.
-        # Rails 7.1 supports composite primary keys so we can implement universal handling. See [FeaturesPlan].
-        # Now to avoid complications, just sweep it under the rag.
-        FeaturesPlan.where(feature_id: ar_object.feature_id, plan_id: ar_object.plan_id).delete_all
-        return []
-      end
-
```

Comment:
as a side effect of supporting `#delete_all` for associations, now we can rid of this ugly hack

Reply:
btw we also have `group_sections` without an `id` column. idk why it was never a problem. Either case, now both shouldn't be as they have `dependent: :delete_all`